Ab#61247

lardo-de-arnaud opened the pull request:
the line

```
auth_object=`curl "https://${host}/api/v1/accounts/1/authentication_providers"  -H "Authorization: Bearer ${token}" -s \
  -F "auth_type=saml"\
  -F "log_out_url=${log_out_url}"\
  -F "idp_entity_id=${idp_entity_id}"\
  -F "log_in_url=${log_in_url}"\
  -F "certificate_fingerprint=${certificate_fingerprint}" | jq` || (printf "Unable to add AzAD auth method\n\n"; exit 1)

```

is this going to work, ie, if the curl fails, will it throw the error message? I'm unsure cos I've used back quotes `

lardo-de-arnaud committed:
AB#61247 Script to add an AzAD auth route to Canvas

Pokes Canvas to create an AzAD auth route.

We also need to modify the login page URL - this is noted in the script as a 'TO DO'

lardo-de-arnaud committed:
Update setup-azad-auth_provider.sh

add curl cmd to set the discovery page to the recently created auth method

lardo-de-arnaud committed:
Update setup-azad-auth_provider.sh

added error handling

buckett commented on setup-azad/setup-azad-auth_provider.sh:
```diff
@@ -0,0 +1,51 @@
+#!/bin/bash
+# This create an AzAD authentication provider in Canvas 
+# typically this is used for Canvas Beta which will have a URL of /login/saml/139 but could be used on any server 
+
```
I'd normally have a `set -e` so it fails on any error, is there a reason not to for this script?

buckett commented on setup-azad/setup-azad-auth_provider.sh:
```diff
@@ -0,0 +1,51 @@
+  -F "log_in_url=${log_in_url}"\
+  -F "certificate_fingerprint=${certificate_fingerprint}" | jq` || (printf "Unable to add AzAD auth method\n\n"; exit 1)
+
+printf "Auth object"
+printf "${auth_object}" | jq
```
Do we still want this in here when it's being run from `cron` as it will mean we get a mail even if there isn't a problem?

buckett replied:
Discussed on chat, plan is to leave them in for now.

buckett reviewed (commented):
Did you say that you still needed to make more changes to this?

lardo-de-arnaud committed:
Update setup-azad-auth_provider.sh

Added code to delete existing production AzAD auth provider as this appears to stop AzAD working on Test / Beta.

lardo-de-arnaud committed:
Update setup-azad-auth_provider.sh

added set -e

lardo-de-arnaud committed:
AB#61247 updated readmes & script & folder names

Added README.md, updated parent README & gave things better names

sebastianchristopher reviewed (approved):
LGTM
